### ISSUE TITLE
[BUG] New content parser

### ISSUE THREAD

**Issue by nlemoine** [bolt/bolt#7507]
When when setting `setcontent_legacy: false`, the following notice is raised: 
```
ContextErrorException in Frontend.php line 170: 
Notice: Undefined property: Bolt\Storage\Query\QueryResultset::$id
```

Details
-------

 - Relevant Bolt Version: 3.5.3
 - Install type: Composer install
 - PHP version: 7.1
 - Used web server: Apache 2.4

Reproduction
------------

### Bug summary

I tried to dig a bit further to figure out where this issue could come from.
- On the [frontend controller](https://github.com/bolt/bolt/blob/v3.5.3/src/Controller/Frontend.php#L152), no content is actually retrieved from the database but it's still passing the [404 check](https://github.com/bolt/bolt/blob/v3.5.3/src/Controller/Frontend.php#L160) because an empty `QueryResultset` object is returned. A [bit later](https://github.com/bolt/bolt/blob/v3.5.3/src/Controller/Frontend.php#L170) in the process, a notice is thrown a there's no `id` property.  
- The page exists but `getContent` is unable to fetch it. Keeping searching, I ended up in the `SelectQueryHandler`.  [`whitelistParameters` returns `false`](https://github.com/bolt/bolt/blob/v3.5.3/src/Storage/Query/Handler/SelectQueryHandler.php#L36) and prevents the query to actually happen because there's a [`log_not_found`](https://github.com/bolt/bolt/blob/v3.5.3/src/Controller/Frontend.php#L152) param in `$params`. Remove it and it works.

At that point, I have not enough knowledge on Bolt internals to fix it and submit a PR. Hence this issue. 

### Steps to reproduce

- On a fresh bolt website (`composer create-project...`) with some pages, set `setcontent_legacy` to `false` in `app/config.yml`. 
- Check a single record page.

### Expected result

Content fetched and displayed with no error.

### Actual result

No content fetched.

**Comment by rossriley** [comment]
Yes, I know about this one and it's already fixed in the mega PR #7291.

There's quite a few bugs fixed in that PR so until it lands I'd recommend not using the flag for legacy off since it's not really stable enough. As soon as that lands most of these bugs will disappear and it will be ready for some more indepth testing.

**Comment by nlemoine** [comment]
Thanks for your reply @rossriley, I'll close the issue and will reopen it if necessary.
